### PR TITLE
Create pack_cwls.yml

### PR DESCRIPTION
**Pull request by ionox0**
Used a GitHub action to pack all workflows after they are changed.

@rhshah

### CODE REVIEW

**Comment by ionox0** [comment]
Not sure if you wanted me to make this for PR's to develop as well as commits to develop, if not I can remove the pull request automation and keep it to just commits to develop

File: .github/workflows/pack_cwls.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          python-version: ${{ matrix.python-version }}
+      - name: Install cwltool
+        run: |
+          python -m pip install toil-msk[cwl]
+      - name: Run cwltool --pack for all workflows
+        run: |
+          echo Removing old packed workflows
+          rm -rf ./packed_workflows/*
+          echo Packing CWL files...
+          find . -name '*.cwl' ! -path '*/command_line_tools/*' ! -path '*/packed_workflows/*' -type f -print0 | xargs -0 -I file sh -c 'base=$(basename file); cwltool --pack file > packed_workflows/${base/.cwl/__packed.cwl}'
```
Comment (rhshah) [inline .github/workflows/pack_cwls.yml:34]: Looks good, what happens when it fails. Also, should we do `cwltool --validate __packed.cwl`

**Review by rhshah — COMMENTED** [review]
Please let me know your thoughts on the comments.

**Comment by rhshah** [comment]
> Not sure if you wanted me to make this for PR's to develop as well as commits to develop, if not I can remove the pull request automation and keep it to just commits to develop

I think PR is good.

File: .github/workflows/pack_cwls.yml (hunk 1 above)
Comment (ionox0) [reply .github/workflows/pack_cwls.yml:34]: The check will fail and the packed file for that workflow will not be created:

Screen Shot 2020-09-10 at 11 55 27 AM

I've also added a validate step afterwards

File: .github/workflows/pack_cwls.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install cwltool
+        run: |
+          python -m pip install toil-msk[cwl]
+      - name: Run cwltool --pack for all workflows
+        run: |
+          echo Removing old packed workflows
+          rm -rf ./packed_workflows/*
```
[the remark this answers is not in the document]
Comment (ionox0) [reply .github/workflows/pack_cwls.yml:32]: I reviewed your PR for that and yep I can make that change after it is merged

**Comment by rhshah** [comment]
@ionox0 just fyi I have merged #145  pull request

**Comment by ionox0** [comment]
@rhshah I had to make a few fixes to get the `--validate` to work due to recent changes of the IDs in the commandlinetools

In addition I had to upgrade to base toil==4.2.0 because toil-msk==3.21.0 is currently using an outdated version of cwltool that errors for the bqsr subworkflow (but the bug was fixed in a later version of cwltool).

These subworkflows are now passing the --pack and --validate steps now using databiosphere toil 4.2.0

4.2.0 also has Nikhil's LSF stability fixes, and the fix for ResolveIndirect as a longer `waitDuration` for the lsf batchSystem https://github.com/DataBiosphere/toil/blob/master/src/toil/batchSystems/lsf.py#L299

Let me know how this all sounds to you. I can perform additional testing with toil 4.2.0, or I can ignore the --validate errors for now and stick with toil-msk==3.21.0

**Comment by rhshah** [comment]
@ionox0 i think I resolved this branch and merged it to the develop can you just check that part

**Pull request merged**